src/System.Management.Automation/FormatAndOutput/common/PSStyle.cs

```diff
@@ -348,29 +413,132 @@ public sealed class FileInfoFormatting
+            /// Custom dictionary handling validation of extension and content.
+            /// </summary>
+            public sealed class FileExtensionDictionary
+            {
+                private static string ValidateExtension(string extension)
+                {
+                    if (!extension.StartsWith('.'))
+                    {
+                        throw new ArgumentException(PSStyleStrings.ExtensionNotStartingWithPeriod);
+                    }
+
+                    return extension;
+                }
+
+                private readonly Dictionary<string, string> _extensionDictionary = new(StringComparer.OrdinalIgnoreCase);
+
+                /// <summary>
+                /// Add new extension and decoration to dictionary.
+                /// </summary>
+                /// <param name="extension">Extension to add.</param>
+                /// <param name="decoration">ANSI string value to add.</param>
+                public void Add(string extension, string decoration)
+                {
+                    _extensionDictionary.Add(ValidateExtension(extension), ValidateNoContent(decoration));
+                }
+
+                /// <summary>
+                /// Remove an extension from dictionary.
+                /// </summary>
+                /// <param name="extension">Extension to remove.</param>
+                public void Remove(string extension)
+                {
+                    _extensionDictionary.Remove(ValidateExtension(extension));
+                }
+
+                /// <summary>
+                /// Clear the dictionary.
+                /// </summary>
+                public void Clear()
+                {
+                    _extensionDictionary.Clear();
+                }
+
+                /// <summary>
+                /// Gets or sets the decoration by specified extension.
+                /// </summary>
+                /// <param name="extension">Extension to get decoration for.</param>
+                /// <returns>The decoration for specified extension.</returns>
+                public string this[string extension]
+                {
+                    get
+                    {
+                        return _extensionDictionary[ValidateExtension(extension)];
+                    }
+
+                    set
+                    {
+                        _extensionDictionary[ValidateExtension(extension)] = ValidateNoContent(value);
+                    }
+                }
+
+                /// <summary>
+                /// Gets whether the dictionary contains the specified extension.
+                /// </summary>
+                /// <param name="extension">Extension to check for.</param>
+                /// <returns>True if the dictionary contains the specified extension, otherwise false.</returns>
+                public bool ContainsKey(string extension)
+                {
+                    return _extensionDictionary.ContainsKey(extension);
```

Comment:
Why don't we use ValidateExtension() here?

Reply:
I'll put it back, just need to cover empty string case where exe's don't have extension on Linux